test/Shared/UserManagerTestBase.cs

```diff
@@ -592,8 +591,8 @@ protected TRole CreateRole(string namePrefix = "")
             {
                 IdentityResultAssert.IsSuccess(await mgr.CreateAsync(u));
             }
-            var usersQ = mgr.Users.Where(u => u.UserName.StartsWith("CanFindUsersViaUserQuerable"));
-            Assert.Null(mgr.Users.FirstOrDefault(u => u.UserName == "bogus"));
+            var usersQ = mgr.Users.Where(u => mgr.GetUserNameAsync(u).Result.StartsWith("CanFindUsersViaUserQuerable"));
+            Assert.Null(mgr.Users.FirstOrDefault(u => mgr.GetUserNameAsync(u).Result == "bogus"));
```

Comment:
If we really think these test are general purpose and not EF only, e.g. if we think they have value for store providers that are not based on EF but support IQueryable, I propose that:

1. The tests for the IQueryable interfaces shouldn't fail if the IQueryable properties are not supported by the store, e.g. the test should test for the capability first.

2. The LINQ implementations are built to recognize simple patterns such as property accesses on the entities, and not calls to Identity. Since the test base class cannot depend on a specific set of entity types, we need to abstract this. We can do it easily by adding a set of virtual methods can invoke to build the expression that needs to be passed to the LINQ operators, e.g. for this case (brain-compiled):

``` C#
protected abstract Expression<Func<TUser, bool>> GetUserNamePredicate(string userName) 
{
}
``` 

An implementation for EF would be
``` C#
protected abstract Expression<Func<IdentityUser, bool>> GetUserNamePredicate(string userName) 
{
    return u => u.NormalizedUserName == userName;
}
```